fastcrypto-zkp/src/api.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use std::io::Cursor;
+
+use ark_bls12_381::Bls12_381;
+use ark_groth16::{Proof, VerifyingKey};
+use ark_serialize::CanonicalDeserialize;
+use fastcrypto::error::FastCryptoError;
+
+use crate::conversions::BlsFr;
+use crate::verifier::{process_vk_special, verify_with_processed_vk, PreparedVerifyingKey};
+
+/// Deserialize bytes as a Arkwork representation of a verifying key, and return a serialized prepared verified key.
+pub fn prepare_pvk_bytes(vk_bytes: Vec<u8>) -> Result<Vec<Vec<u8>>, FastCryptoError> {
+    let reader_vk = Cursor::new(vk_bytes);
+    let vk = VerifyingKey::<Bls12_381>::deserialize(reader_vk)
+        .map_err(|_| FastCryptoError::InvalidInput)?;
+
+    process_vk_special(&vk).as_serialized()
+}
+
+/// Verify Groth16 proof using the serialized representation of verifying key, input and proof.
+pub fn verify_groth16_in_bytes(
+    pvk_bytes: Vec<Vec<u8>>,
+    x_bytes: Vec<u8>,
```

Comment:
I'm personally ok with the 3 different inputs mainly for readability and deserialization complexity, but let's rename `x` to something meaningful (public_input?).